backend/frontend/src/screens/UpdateProfileScreen.js

```diff
@@ -107,16 +118,25 @@ function UpdateprofileScreen() {
 
   useEffect(() => {
     if (userUpdateSuccess && submit) {
-      toast.success("user details updated");
+      toast.success("User details updated");
       setSubmit(false);
     } else if (userUpdateError && submit) {
-      toast.error("incorrect OTP");
-      setOpenForm(false);
+      setOtp("");
+      setOtpError(true);
+      setOtpValid(false);
+      setIncorrectAttempts(incorrectAttempts + 1);
+      if (incorrectAttempts >= 3) {
+        setOpenForm(false);
+        toast.error("Too many invalid attempts, try again later");
+      } else {
+        toast.error("Incorrect OTP");
+      }
+      setSubmit(false);
     }
     dispatch({
       type: USER_UPDATE_PROFILE_RESET,
     });
-  }, [dispatch, navigate, submit, userUpdateError, userUpdateSuccess]);
+  }, [dispatch, navigate, submit, userUpdateError, userUpdateSuccess, incorrectAttempts]);
 
```

Comment:
Dependencies need to be in Alpabetical order.

Reply:
done